packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2756,8 +2786,23 @@ abstract class Element implements BuildContext {
   Set<InheritedElement> _dependencies;
   bool _hadUnsatisfiedDependencies = false;
 
+  void _debugCheckStateIsActiveForAncestorLoopkup() {
+    assert(() {
+      if (_debugLifecycleState != _ElementLifecycle.active) {
+        throw new FlutterError(
+          'Looking up a deactivated widget\'s ancestor is unsafe.\n'
+          'At this point the state of the widget\'s element tree is no longer '
+          'stable. To safely refer to a widget\'s ancestor in its dispose() method, '
+          'save a reference to the ancestor in the widget\'s initState() method.\n'
```

Comment:
this message needs fixing just like you fixed the comments above -- calling inheritFromWidgetOfExactType in didChangeDependencies and all that, rather than initState.